File: languages/GBL/Haskell/Language/GBL/Sample.hs
-- BEGIN ...
module Language.GBL.Sample where
import Data.IORef
import Language.GBL.Syntax
-- END ...
mkSmallWorld :: IO World
mkSmallWorld = do
  r1 <- newIORef undefined
  r2 <- newIORef undefined
  writeIORef r1 Person { getName = "joe", getBuddy = Just r2 }
  writeIORef r2 Person { getName = "bill", getBuddy = Just r1 }
  return $ World [r1, r2]

File: languages/BL/Java/org/softlang/tbl/Sample.java
// BEGIN ...
package org.softlang.tbl;
import org.softlang.tbl.Syntax.*;
// END ...
public class Sample {
    public World smallWorld() {
	World w = new World();
        Person p1 = new Person();
	Person p2 = new Person();
	w.persons = new Person[] {p1, p2};
        p1.name = "joe";
        p1.buddy = "bill";
        p2.name = "bill";
        p2.buddy = "joe";
        return w;
    }
}
